rb_tree.py

```diff
@@ -9,48 +9,55 @@
 
 
 class Node:
+
     def __init__(self, value, color, parent, left=None, right=None):
+        
         self.value = value
         self.color = color
         self.parent = parent
         self.left = left
         self.right = right
 
     def __repr__(self):
+        
         return '{color} {val} Node'.format(color=self.color, val=self.value)
 
     def __iter__(self):
+        
+        print(str(self.value) + " (" + str(self.color)+")")
+        
         if self.left.color != NIL:
+            print('left ({})'.format(str(self.value)));
             yield from self.left.__iter__()
 
-        yield self.value
-
         if self.right.color != NIL:
+            print('right ({})'.format(str(self.value)));
             yield from self.right.__iter__()
+ 
 
     def __eq__(self, other):
-        if self.color == NIL and self.color == other.color:
-            return True
+            
+            if self.color == NIL and self.color == other.color:
```

Comment:
Does this even compile? There is too much space (8 spaces) in this method body

Reply:
It was compiling but it was a mistake. Now, fixed.